<PerfMonitor/> - rendering stats (#873)
threlte/threlte: apps/docs/src/content/reference/extras/perf-monitor.mdx

---
{
  order: 10.5,
  category: '@threlte/extras',
  name: '<PerfMonitor>',
  sourcePath: 'packages/extras/src/lib/components/PerfMonitor/PerfMonitor.svelte',
  type: 'component',
  'componentSignature':
    {
      'props':
        [
					 {
            name: 'logsPerSecond',
            type: 'number',
            default: '10',
            required: false,
            'description': 'Refresh rate of the logs.'
          	},
						{
            name: 'showGraph',
            type: 'boolean',
            default: 'true',
            required: false,
            'description': 'Toggles cpu/gpu/fps graphs rendering.'
          	},
						{
            name: 'memory',
            type: 'boolean',
            default: 'true',
            required: false,
            'description': 'Toggles memory info visiblity (geometries, textures, shaders)'
          	},
						{
            name: 'enabled',
            type: 'boolean',
            default: 'true',
            required: false,
            'description': 'Toggles stats collection.'
          	},
						{
            name: 'visible',
            type: 'boolean',
            default: 'true',
            required: false,

>             'description': 'Toggles visibility of the monitor html element. Disabling this does not stop collection of the stats.'

Reviewer: ```suggestion
            'description': 'Toggles visibility of the monitor html element. Setting false does not stop collection of the stats.'
```

          	},
						{
            name: 'actionToCallUI',
            type: 'string',
            default: ' ',
            required: false,
            'description': 'By default is disabled [''], but if set and type this text in the tab window three-perf dev gui will be shown.'
          	},
						{
            name: 'guiVisible',
            type: 'boolean',
            default: 'false',
            required: false,
            'description': 'Three-perf dev gui visiblity.'
          	},
						{
            name: 'backgroundOpacity',
            type: 'number',
            default: '0.7',
            required: false,
            'description': 'Stats block background opacity level.'
          	},
						{
            name: 'scale',
            type: 'number',
            default: '1',
            required: false,
            'description': 'Scale of the stats block html element.'
          	},
						{
            name: 'anchorX',
            type: "'left' | 'right'",

>             default: '[]',

Reviewer: ```suggestion
            default: 'left',
```

            required: false,
            'description': 'Stats html element horizontal anchor.'
          	},
						{
            name: 'anchorY',
            type: "'top' | 'bottom'",

>             default: '[]',

Reviewer: ```suggestion
            default: 'top',
```

            required: false,
            'description': 'Stats html element vertical anchor.'
          	},
        ]
    }
}
---

> Utility component for monitoring basic rendering statistics. Based on [three-perf](https://github.com/TheoTheDev/three-perf).

Reviewer 2: ```suggestion
Utility component for monitoring basic rendering statistics using [three-perf](https://github.com/TheoTheDev/three-perf).
```

<Example path="extras/perf-monitor" />

## Usage

Simply drop in the `<PerfMonitor>` component as a child of Threlte `<Canvas>`.

> The component starts measuring before `mainStage` and ends at `renderStage`.

Reviewer 2: ```suggestion
The component starts measuring before `mainStage` and ends after `renderStage`.
```

> Learn more about the stages [here](/docs/learn/basics/scheduling-tasks)

Reviewer 2: ```suggestion
Learn more about tasks in threlte [here](/docs/learn/basics/scheduling-tasks)
```
